### ISSUE TITLE
Keyless Authentication to support Application Default Login

### ISSUE THREAD

**Issue by harishatk** [google-github-actions/auth#107]
### TL;DR

Gcloud Application Default login only allowed with the user credentials or the libraries will be authenticated based on the ADC (run platform/if serviceAccount key in the GOOGLE_APPLICATION_CREDENTIALS).  The general practice in CI (git Actions) for a Java application build (for specific actions that demands access to GCP like JAR/package build and export, unit tests to start the application and utilize cloud resources, etc.,) Google ServiceAccount Key is retrieved from a secure location (usually Git Secrets) and pushed to the build container in the path specified in env GOOGLE_APPLICATION_CREDENTIALS (with export_default_credentials as true in google-github-actions/setup-gcloud@master). 

But after switching to the workload_identity_provider in google-github-actions/auth@v0, only the credentials can be exported and which the application libraries (either mvn nor graddle nor npm, etc.,) can't recognize the ADC and fails to login.

### Detailed design

```markdown
The ask is to also allow the application login (Google application libraries from maven/graddle/npm) with the keyless authentication, and the same I hope can be achieved either via
1. Updating the Gcloud auth application-default login to allow and accept the credentials outputed from the google-github-actions/auth@v0
(or)
2. Create a shortlived(temporary) JSON Key for the ServiceAccount which is used to authorize for the token exchange and to keep it build container that can be recognized by ADC ( and to delete the ServiceAccount key on build termination)
(or)
3. Provide a mechanism to support application login in CI steps (for eg., the gcloud auth login is allowed with the credentials outputed from the auth step)
```

**Comment by harishatk** [comment]
Thanks for the quick reply and the suggestion. I could see the application always picked the default Project ID as  WIF defined GCP ID, that caused the failure. And your comments helped me to understand the error statement better and I could reason for failure as the application fails for a resource access that was getting rejected for as it looking in a wrong project-id (where the WIF is defined and setup) even though passing mvn command line option "-DGOOGLE_CLOUD_PROJECT"  with the required project-id didn't help, but luckily by setting the env variable "GOOGLE_CLOUD_PROJECT" to the desired project-id fixed the issue :) Thanks again for your help.
